### PR TITLE
React base

### PR DESCRIPTION
**Pull request by moridyn**
Simple react app which only renders menu

### CODE REVIEW

File: example_project/react-app/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+To start project you need node version 6 or more
```
Comment (FinalAngel) [inline example_project/react-app/README.md]: we normally use restructured text for all readme's

File: example_project/react-app/public/index.html [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<!doctype html>
+<html lang="en">
+  <head>
```
Comment (FinalAngel) [inline example_project/react-app/public/index.html]: always 4 spaces

File: example_project/react-app/src/App.css [hunk 3]
```diff
@@ -0,0 +1,20 @@
+.App {
```
Comment (FinalAngel) [inline example_project/react-app/src/App.css]: lowercase

File: example_project/react-app/src/App.css [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    text-align: center;
+}
+
+.App-logo {
+    animation: App-logo-spin infinite 20s linear;
```
Comment (FinalAngel) [inline example_project/react-app/src/App.css]: also lowercase for variables or names

File: example_project/requirements.txt [hunk 5]
```diff
@@ -1,4 +1,5 @@
-django-cms<3.3
+django-cms==3.4
+djangorestframework==3.5.2
 django-treebeard>=4.0
```
Comment (FinalAngel) [inline example_project/requirements.txt:3]: you need to add `django-cors-headers` as well to the requirements

File: example_project/README.md [hunk 6]
```diff
@@ -0,0 +1,8 @@
+To start project you need to:
+ - create virtualenv with: virtualenv venv
+ - activate env with: source venv/bin/activate
+ - install requirement with: pip install -r requirements.txt
+ - call migrations: ./manage.py migrate
```
Comment (FinalAngel) [inline example_project/README.md]: first you need to create a superuser `python manage.py createsuperuser`

File: example_project/README.md [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+ - run dev server: ./manage.py runserver
```
Comment (FinalAngel) [inline example_project/README.md]: also mention that you can access the api via `http://localhost:8000/en/api/`

Comment (FinalAngel) [inline example_project/README.md]: this starts the django runserver where I can add pages etc. how can I access the react app?

**Review by FinalAngel — CHANGES_REQUESTED** [review]

File: example_project/README.md (hunk 7 above)
Comment (FinalAngel) [reply example_project/README.md]: Integrate the README from the react_app directly here, noting:

```
- open a new terminal window
- navigate to the react-app
- run npm install
- run npm start

```

and then the additional infos regarding localserver etc from the react-app's readme

File: example_project/react-app/README.md [hunk 8]
```diff
@@ -0,0 +1,8 @@
...
+Call commands:
+ - npm install
+ - npm start
```
Comment (FinalAngel) [inline example_project/react-app/README.md]: this throws an error for me:

```
⋊> ~/S/d/e/react-app on react_base  npm start                           08:38:59

> react-app@0.1.0 start /Users/finalangel/Sites/djangocms-rest-api/example_project/react-app
> react-scripts start

/Users/finalangel/Sites/djangocms-rest-api/example_project/react-app/node_modules/react-scripts/scripts/start.js:46
var isSmokeTest = process.argv.some(arg => arg.indexOf('--smoke-test') > -1);
                                        ^^
SyntaxError: Unexpected token =>
    at exports.runInThisContext (vm.js:73:16)
    at Module._compile (module.js:443:25)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:501:10)
    at startup (node.js:129:16)
    at node.js:814:3

npm ERR! Darwin 16.1.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "start"
npm ERR! node v0.12.4
npm ERR! npm  v2.11.3
npm ERR! code ELIFECYCLE
npm ERR! react-app@0.1.0 start: `react-scripts start`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the react-app@0.1.0 start script 'react-scripts start'.
npm ERR! This is most likely a problem with the react-app package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     react-scripts start
npm ERR! You can get their info via:
npm ERR!     npm owner ls react-app
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/finalangel/Sites/djangocms-rest-api/example_project/react-app/npm-debug.log
```

Comment (vxsx) [reply example_project/react-app/README.md]: you have old node :)
try with node 6

Comment (FinalAngel) [reply example_project/react-app/README.md]: Yes, eventually note the requirements of the node / npm version

File: example_project/requirements.txt (hunk 5 above)
Comment (FinalAngel) [reply example_project/requirements.txt:3]: @moridyn  missed one here

File: example_project/README.md [hunk 9]
```diff
@@ -0,0 +1,32 @@
+==============================
+djangocms-rest-api example app
+==============================
+
+Test application for Django Cms Rest Api
```
Comment (FinalAngel) [inline example_project/README.md:5]: django CMS Rest API

File: example_project/README.md [hunk 10]
```diff
@@ -0,0 +1,32 @@
+==============================
```
Comment (FinalAngel) [inline example_project/README.md:1]: please use .rst as default format

File: example_project/README.md [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+djangocms-rest-api example app
+==============================
+
+Test application for Django Cms Rest Api
+
+Django app
+----------
+
+* create virtualenv with: virtualenv venv
+* activate env with: source venv/bin/activate
```
Comment (FinalAngel) [inline example_project/README.md:11]: mark them as code via ` ` > ``virtualenv venv``

File: example_project/README.md [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+* install requirement with: pip install -r requirements.txt
+* call migrations: ./manage.py migrate
+* create a superuser ./manage.py createsuperuser
+* run dev server: ./manage.py runserver
+ 
+You can access api via http://localhost:8000/en/api/
+
```
Comment (FinalAngel) [inline example_project/README.md:18]: additional space

**Review by FinalAngel — CHANGES_REQUESTED** [review]
@vxsx wdyt?

**Review by vxsx — COMMENTED** [review]
looks good overall

**Pull request merged**